File: client/csharp_example/Test/SimpleTest.cs
using AUTD3Sharp;
using System;

namespace AUTD3SharpTest.Test
{
    class SimpleTest
    {
        public static void Test()
        {
            Console.WriteLine("Start Simple Test");

            float x = 83.0f;
            float y = 66.0f;
            float z = 150.0f;

            using (var autd = new AUTD())
            {
                autd.Open();
                autd.AddDevice(Vector3f.Zero, Vector3f.Zero);

                autd.AppendModulationSync(AUTD.SineModulation(150)); // AM sin 150 HZ
                autd.AppendGainSync(AUTD.FocalPointGain(x, y, z)); // FocalPoint at (x,y,z)

                Console.WriteLine("press any key to finish...");
                Console.ReadKey(true);
            }
        }

        private SimpleTest() { }
    }
}

File: future/autd/examples/tests/simple_test.rs
/*
 * File: simple_test.rs
 * Project: example
 * Created Date: 12/12/2019
 * Author: Shun Suzuki
 * -----
 * Last Modified: 06/02/2020
 * Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
 * -----
 * Copyright (c) 2019 Hapis Lab. All rights reserved.
 *
 */

use autd::gain::*;
use autd::modulation::*;
use autd::prelude::*;

pub fn simple_test(ifname: &str, link_type: LinkType) -> Result<AUTD, failure::Error> {
    let mut autd = AUTD::create();

    autd.add_device(Vector3f::zeros(), Vector3f::zeros());

    autd.open(ifname, link_type)?;

    let g = FocalPointGain::create(Vector3f::new(90., 70., 150.));
    autd.append_gain_sync(g);

    let m = SineModulation::create(150);
    autd.append_modulation_sync(m);

    Ok(autd)
}
